onedal/datatypes/_data_conversion.py

```diff
@@ -18,28 +18,62 @@
 
 import numpy as np
 
-from daal4py.sklearn._utils import make2d
 from onedal import _backend, _is_dpc_backend
 
-from ..utils import _is_csr
-
 
 def _apply_and_pass(func, *args, **kwargs):
     if len(args) == 1:
         return func(args[0], **kwargs)
     return tuple(map(lambda arg: func(arg, **kwargs), args))
 
 
-if _is_dpc_backend:
+def _convert_one_to_table(arg):
+    # All inputs for table conversion must be array-like or sparse, not scalars
+    return _backend.to_table(np.atleast_2d(arg) if np.isscalar(arg) else arg)
+
+
+def to_table(*args):
+    """Create oneDAL tables from scalars and/or arrays.
+
+    Note: this implementation can be used with contiguous scipy.sparse, numpy
+    arrays, sycl_usm_ndarrays, and scalars. Tables will use pointers to the
+    original array data. Scalars will be copies. Arrays may be modified in-
+    place by oneDAL during computation. This works for data located on CPU and
+    SYCL-enabled Intel GPUs. Each array may only be of a single datatype (i.e.
+    each must be homogeneous).
+
+    Parameters
+    ----------
+    *args : {scalar, numpy array, sycl_usm_ndarray, csr_matrix, or csr_array}
+        arg1, arg2... The arrays should be given as arguments.
 
-    from ..utils._dpep_helpers import dpctl_available, dpnp_available
+    Returns
+    -------
+    tables: {oneDAL homogeneous tables}
+    """
+    return _apply_and_pass(_convert_one_to_table, *args)
 
-    if dpctl_available:
-        import dpctl.tensor as dpt
 
-    if dpnp_available:
+if _is_dpc_backend:
+
+    try:
```

Comment:
Could you please put logs for circular import issue here that you have mentioned?

Reply:
Done. Please check @samir-nasibli